src/app/layout.js

```diff
@@ -1,18 +1,16 @@
 import React from 'react'
-import { ToastContainer } from 'react-toastify'
-import 'react-toastify/dist/ReactToastify.css'
+import { FirebaseAuthProvider } from '@/contexts/firebaseAuthContext'
 
 export const metadata = {
-  title: '#',
-  description: '#',
+  title: 'Main root of the Project',
+  description: 'Main root description of the Project',
 }
 
 export default function AppLayout({ children }) {
   return (
     <html lang="en" className={`bg-white`}>
       <body className="flex flex-col">
-        {children}
-        <ToastContainer />
+        <FirebaseAuthProvider>{children}</FirebaseAuthProvider>
```

Comment:
NIT: can you test the page load time with that, perhaps we can optimize that if we find a way to lazy load auth